internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go

```diff
@@ -46,13 +44,13 @@ const (
 
 // Downloader is a downloader able to fetch artifacts from elastic.co web page.
 type Downloader struct {
-	log    progressLogger
```

Comment:
Nit: I always prefer the logger as interfaces for mocking and loose coupling, instead of relying on a more concrete type

Reply:
In general, I agree but I'm also a fan of YAGNI.  We can easily change this back to an interface if/when we need it, i.e. have more than one implementation.